dkg/nodesigs.go

```diff
@@ -26,20 +28,30 @@ func nodeSigMsgIDs() []string {
 
 // nodeSigBcast handles broadcasting of K1 signatures over the lock hash via the bcast protocol.
 type nodeSigBcast struct {
-	otherSigs   chan *dkgpb.MsgNodeSig
+	otherSigs   sync.Map
```

Comment:
nit: Add a comment that the key value types are

Reply:
also, what about refactoring this as a fixed size slice, so basically peerIdx as keys? Then we do not need to convert from map the a slice, since it is already correct. We also dedup using peerIdx keys. Only thing is we would need a special function to check is all sigs populated
```
func (b nodeSigBcast) allsigs() bool {
 b.mu.Lock()
 defer unlock
 
 for sig := range b.sigs {
     if sig == nil { return false }
 } 
 
 return true
}
```